client/jetpack-onboarding/main.jsx

```diff
@@ -113,8 +132,27 @@ export default connect(
 		const siteId = getUnconnectedSiteIdBySlug( state, siteSlug );
 		const settings = getJetpackOnboardingSettings( state, siteId );
 		const isBusiness = get( settings, 'siteType' ) === 'business';
-		const isRequestingSettings = getRequest( state, requestJetpackOnboardingSettings( siteId ) )
-			.isLoading;
+
+		const isRequestingWhetherConnected =
+			isRequestingSite( state, siteId ) || isRequestingSites( state );
+		const isConnected = isJetpackSite( state, siteId );
+
+		let jpoAuth;
```

Comment:
With approach of having the logic to decide whether to pass onboarding auth credentials to the actions placed in the component, we lack tests for this logic. I think it is pretty important to have tests for this, as it's essential for allowing JPO requests to work with both connected and unconnected sites. Do you think we can add some component tests for this? Could happen in a subsequent PR, of course.

Reply:
Testing `connect()`ed components used to be a bit of a pain in the past (though I haven't revisited that sort of problem in a while). If we go with a separate selector as you suggest [here](https://github.com/Automattic/wp-calypso/pull/22775#discussion_r171494411), we might be able to cover that with unit tests for all these cases.